src/material/form-field/form-field.ts

```diff
@@ -701,14 +701,23 @@ export class MatFormField
   /** Checks whether the form field is attached to the DOM. */
   private _isAttachedToDom(): boolean {
     const element: HTMLElement = this._elementRef.nativeElement;
+    const shadowRoot = _getShadowRoot(element);
```

Comment:
This should move to before the `return` since it's not used in the case where `getRootNode` exists.